cmux-tui/bindings/cpp/src/unix_transport.cpp

```
@@ -322,19 +591,34 @@ void UnixTransport::close() noexcept {
+    if (runtime_socket_path_fits(fallback_base, session)) {
+        return runtime_socket_path(fallback_base, session);
     }
-    return preferred;
+    auto hashed = hashed_session_socket_path(session);
+    if (!unix_socket_path_fits(hashed)) {
+        return make_error(ErrorCode::invalid_argument, "derived Unix socket path is too long");
+    }
+    return hashed;
+}
+
+std::string default_socket_path(std::string_view session) {
```

Cppcheck (2.21.0)

[style] The function 'default_socket_path' is never used.

(unusedFunction)